bofire/strategies/strategy.py

```diff
@@ -242,7 +232,7 @@ def ask(
                 ), "Number of requested candidates is larger than the pool from which they should be chosen."
             candidates = self._choose_from_pool(candidate_pool, candidate_count)
 
-        self.domain.validate_candidates(candidates=candidates)
+        self.domain.validate_candidates(candidates=candidates, only_inputs=True)
```

Comment:
I would like to have it here always with `only_inputs = False`, as this is our general definition of candidates, that ones always also has the predicted value, uncertainty and objective value. In case of the random strategy this would be `nan`.

Reply:
We have `Strategy` and `PredictiveStrategy`. I would argue that only predicitve strategies need to have predictions, whereas non-predictive strategies such as random sampling and certain classic experimental designs don't. 

Can you give me a point where we would run into problems with candidates without output predictions?

Reply:
You are right. Only point from my side would be that our backend expects it in this way, which is I tink not a valid point ;) 

If we do it as you propose (which is fine for me), then we have to change the `ask` for the predictive strategy in a way that this ones calls it with the additional checks, to make sure that the output of the predictive one has also the predictions.

Reply:
We could implement `PredictiveStrategy.ask` 
```
def ask(...):
    super().ask()
    self.domain.validate_candidates(candidates=candidates)
```

Reply:
Yep, this would be what I prefer. Can you add it to the PR?